lib/spandex.ex

```diff
@@ -323,8 +364,8 @@ defmodule Spandex do
     strategy = opts[:strategy]
 
     case strategy.get_trace(opts[:trace_key]) do
-      {:ok, %Trace{id: trace_id, priority: priority, baggage: baggage, stack: [%Span{id: span_id} | _]}} ->
-        {:ok, %SpanContext{trace_id: trace_id, priority: priority, baggage: baggage, parent_id: span_id}}
+      {:ok, %Trace{id: trace_id, sampling: sampling, baggage: baggage, stack: [%Span{id: span_id} | _]}} ->
+        {:ok, %SpanContext{trace_id: trace_id, priority: sampling.priority, baggage: baggage, parent_id: span_id}}
```

Comment:
If we're extracting `priority` from `sampling` it needs to be described in typespecs. Also why keep `priority` in `SpanContext` but replace it with `sampling` in the trace?